src/Algebra/Graph/Fold.hs

```diff
@@ -150,6 +152,9 @@ instance (Ord a, Show a) => Show (Fold a) where
 instance Ord a => Eq (Fold a) where
     x == y = T.adjacencyMap x == T.adjacencyMap y
 
+instance NFData a => NFData (Fold a) where
+    rnf = foldg () rnf seq seq
```

Comment:
This is not working, because of laziness: it is now unnecessary to evaluate the rest of the foldg when encountering a `connect` or an `overlay`.

Using the same notation than in the issue and your idea as NFData:
```
> let x = fmap primesTo $ clique $ primesTo 1500
(0.01 secs, 408,424 bytes)
> rnf x
()
(0.03 secs, 408,536 bytes)
> vertexCount x
239
(3.35 secs, 7,335,632,352 bytes
```

Sadly the only way to force values is `seq` ^^

Reply:
Ah, I see, indeed :) And what about ``\x y -> x `seq` y `seq` ()``? I'm puzzled by the asymmetry of your solution.

Reply:
I thought the same as you, and here is the benchmarks using your solution:
```
> let x = fmap primesTo $ clique $ primesTo 1500
(0.02 secs, 408,752 bytes)
> rnf x
()
(3.50 secs, 7,206,151,440 bytes)
> vertexCount x
239
(0.02 secs, 543,152 bytes)
```
If you compare with what I gave in the issue, these are almost the same numbers than what is in the PR.
I think it is working for the same reason as `Algebra.Graph`: we force the left branch and return the right, but because rnf needs its result to be `()` he will evaluate the right branch :)

Reply:
> we force the left branch and return the right, but because rnf needs its result to be () he will evaluate the right branch

Oh, I see. That's nice :)